pages/pil/update/index.js

```diff
@@ -33,10 +35,17 @@ module.exports = settings => {
   });
 
   app.use((req, res, next) => {
+
     const values = get(req.session, `form[${req.pilId}].values`);
+
+    const trainingSpecies = flattenDeep(req.model.certificates.map(c => c.modules.map(m => m.species)));
+    const exemptionsSpecies = flattenDeep(req.model.exemptions.map(e => e.species));
+    const species = trainingSpecies.concat(exemptionsSpecies);
+
     req.model = {
       ...req.pil,
-      ...values
+      ...values,
+      species: species.filter((v, i) => species.indexOf(v) === i)
```

Comment:
Is this effectively a uniqueness test? If so could we use something a bit more explicit? i.e. https://lodash.com/docs/4.17.11#uniq

Reply:
done